### DISCUSSION TITLE
Undefined constant GuzzleHttp\Client::VERSION

### DISCUSSION THREAD

**Discussion by SteveMullen** [aws/aws-sdk-php#2442] [Q&A] [answered]
### Describe the bug

PHP Fatal Error

This just started happening today.  Nothing (that I know of) had changed.  I did upgrade from aws.3.198.2 to aws.3.220.3 AFTER the problem began, hoping that would resolve it.  But it did not.

```
[25-Apr-2022 14:37:28 UTC] PHP Fatal error:  Uncaught Error: Undefined constant GuzzleHttp\Client::VERSION in phar:///home/hidden/aws/phar/aws.3.220.3.phar/GuzzleHttp/functions.php:136
Stack trace:
#0 phar:///home/hidden/aws/phar/aws.3.220.3.phar/Aws/Handler/GuzzleV6/GuzzleHandler.php(40): GuzzleHttp\default_user_agent()
#1 phar:///home/hidden/aws/phar/aws.3.220.3.phar/Aws/WrappedHttpHandler.php(87): Aws\Handler\GuzzleV6\GuzzleHandler->__invoke(Object(GuzzleHttp\Psr7\Request), Array)
#2 phar:///home/hidden/aws/phar/aws.3.220.3.phar/Aws/ClientSideMonitoring/AbstractMonitoringMiddleware.php(126): Aws\WrappedHttpHandler->__invoke(Object(Aws\Command), Object(GuzzleHttp\Psr7\Request))
#3 phar:///home/hidden/aws/phar/aws.3.220.3.phar/Aws/S3/PermanentRedirectMiddleware.php(43): Aws\ClientSideMonitoring\AbstractMonitoringMiddleware->__invoke(Object(Aws\Command), Object(GuzzleHttp\Psr7\Request))
#4 phar:///home/hidden/aws/phar/aws.3.220.3.phar/Aws/S3/PutObjectUrlMiddleware.php(41): Aws\S3\PermanentRedirectMiddleware->__invoke(Object(Aws\Command), Object(GuzzleHttp\Psr7\Request))
#5 phar:///home/hidden/aws/phar/aws.3.220.3.phar/Aws/Middleware.php(126): Aws\S3\PutObjectUrlMiddleware->__invoke(Object(Aws\Command), Object(GuzzleHttp\Psr7\Request))
#6 /home/hidden/public_vendor/vendor/guzzlehttp/promises/src/FulfilledPromise.php(41): Aws\Middleware::Aws\{closure}(Object(Aws\Credentials\Credentials))
#7 /home/hidden/public_vendor/vendor/guzzlehttp/promises/src/TaskQueue.php(48): GuzzleHttp\Promise\FulfilledPromise::GuzzleHttp\Promise\{closure}()
#8 /home/hidden/public_vendor/vendor/guzzlehttp/promises/src/Promise.php(248): GuzzleHttp\Promise\TaskQueue->run(true)
#9 /home/hidden/public_vendor/vendor/guzzlehttp/promises/src/Promise.php(224): GuzzleHttp\Promise\Promise->invokeWaitFn()
#10 /home/hidden/public_vendor/vendor/guzzlehttp/promises/src/Promise.php(269): GuzzleHttp\Promise\Promise->waitIfPending()
#11 /home/hidden/public_vendor/vendor/guzzlehttp/promises/src/Promise.php(226): GuzzleHttp\Promise\Promise->invokeWaitList()
#12 /home/hidden/public_vendor/vendor/guzzlehttp/promises/src/Promise.php(62): GuzzleHttp\Promise\Promise->waitIfPending()
#13 phar:///home/hidden/aws/phar/aws.3.220.3.phar/Aws/AwsClientTrait.php(58): GuzzleHttp\Promise\Promise->wait()
#14 phar:///home/hidden/aws/phar/aws.3.220.3.phar/Aws/AwsClientTrait.php(86): Aws\AwsClient->execute(Object(Aws\Command))
#15 /home/hidden/public_rets/__classes/_roverS3.php(652): Aws\AwsClient->__call('listObjects', Array)
```

### Expected Behavior

Correctly get a list of files from an S3 service

### Current Behavior

PHP Fatal Error

### Reproduction Steps

All S3 operations seem to fail.

### Possible Solution

Some forums suggest GuzzleHttp\Client::VERSION is no longer supported, but the call stack makes it look like the GuzzleHttp\Client::VERSION reference is in `aws/phar/aws.3.220.3.phar/GuzzleHttp/functions.php`

### Additional Information/Context

guzzlehttp/guzzle             7.4.2 

### SDK version used

aws.3.220.3

### Environment details (Version of PHP (`php -v`)? OS name and version, etc.)

PHP 8.0.18

**Comment by SteveMullen** [comment]
I got this working by installing the AWS api via composer, and no longer using the straight php api.  I would still recommend this be looked at - for folks using the php api and also have Guzzle installed via Composer.

**Comment by yenfryherrerafeliz** [comment, marked as answer]
Hi @SteveMullen, thank you for reaching out. I tried to reproduce the reported issue with the environment details that you provided and I was not able to, which indicates that the issue could be related with something else, maybe a conflict with different guzzle versions being load, or possibly there was a guzzle version locked down by the composer.lock file which would've required you to run "composer update"; You would've be able to check this by running "composer show guzzlehttp/guzzle" and this would've tell you the guzzle version that is being load, nonetheless I am glad you found a workaround.

For now I will close this issue!

Thank you!

**Reply by gidzr** [reply]
I have come across the same error.
It occurred when I tried to use the PHP SDK to getBundles from Lightsail, when I ran this AWS php example: _https://docs.aws.amazon.com/code-samples/latest/catalog/php-lightsail-GetBundles.php.html_

I've followed it to the letter. 

I was previously using the AWS composer bundles with no problems for S3 and Kinesis functions. It's only been problematic for the Lightsail which seems to build the client+credentials in a slightly different way. All the other clients (s3) have the credentials in the client creation, but Lightsail client doesn't allow this.

When I initially ran getBundles from Lightsail it threw an access error which I fixed by putting my credentials into an c:/user/.aws/credentials file, this fixed the credentials error, but then threw this Guzzle error.

Fatal error: Uncaught Error: Undefined constant GuzzleHttp\Client::VERSION in C:\xampp80\php\vendor\aws\GuzzleHttp\functions.php:136 Stack trace: #0 C:\xampp80\php\vendor\aws\Aws\Handler\GuzzleV6\GuzzleHandler.php(40): GuzzleHttp\default_user_agent() #1 C:\xampp80\php\vendor\aws\Aws\WrappedHttpHandler.php(87): Aws\Handler\GuzzleV6\GuzzleHandler->__invoke(Object(GuzzleHttp\Psr7\Request), Array) #2 C:\xampp80\php\vendor\aws\Aws\ClientSideMonitoring\AbstractMonitoringMiddleware.php(126): Aws\WrappedHttpHandler->__invoke(Object(Aws\Command), Object(GuzzleHttp\Psr7\Request)) #3 C:\xampp80\php\vendor\aws\Aws\Middleware.php(126): Aws\ClientSideMonitoring\AbstractMonitoringMiddleware->__invoke(Object(Aws\Command), Object(GuzzleHttp\Psr7\Request)) #4 C:\xampp80\php\vendor\guzzlehttp\promises\src\FulfilledPromise.php(41): Aws\Middleware::Aws\{closure}(Object(Aws\Credentials\Credentials)) #5 C:\xampp80\php\vendor\guzzlehttp\promises\src\TaskQueue.php(48): GuzzleHttp\Promise\FulfilledPromise::GuzzleHttp\Promise\{closure}() #6 C:\xampp80\php\vendor\guzzlehttp\promises\src\Promise.php(248): GuzzleHttp\Promise\TaskQueue->run(true) #7 C:\xampp80\php\vendor\guzzlehttp\promises\src\Promise.php(224): GuzzleHttp\Promise\Promise->invokeWaitFn() #8 C:\xampp80\php\vendor\guzzlehttp\promises\src\Promise.php(269): GuzzleHttp\Promise\Promise->waitIfPending() #9 C:\xampp80\php\vendor\guzzlehttp\promises\src\Promise.php(226): GuzzleHttp\Promise\Promise->invokeWaitList() #10 C:\xampp80\php\vendor\guzzlehttp\promises\src\Promise.php(62): GuzzleHttp\Promise\Promise->waitIfPending() #11 C:\xampp80\php\vendor\aws\Aws\AwsClientTrait.php(58): GuzzleHttp\Promise\Promise->wait() #12 C:\xampp80\php\vendor\aws\Aws\AwsClientTrait.php(86): Aws\AwsClient->execute(Object(Aws\Command)) #13 C:\xampp80\htdocs\blockvu.com\testaws.php(45): Aws\AwsClient->__call('getBundles', Array) #14 {main} thrown in C:\xampp80\php\vendor\aws\GuzzleHttp\functions.php on line 136

**Reply by gidzr** [reply]
cool - found the reasoning for this error in another discussion:

https://github.com/aws/aws-sdk-php/issues/2006  <--- here

Someone said this error was caused by multiple guzzle scripts running.. and sure enough, I had the AWS guzzle script AND the php guzzle loaded because I also required the main vendor autoload.

"require '../php/vendor/autoload.php'" <--- once I removed this the issue was resolved.

Cheers!

**Reply by JonLaliberte** [reply]
@yenfryherrerafeliz 

Started receiving this error recently - I'm not sure when it started though, probably in the last 6 weeks or so.
We did recently change from PHP 7.3 to 7.4.

The problem was present with v3.274.1, but I updated to 3.275.6 today to test and the problem is still happening.
 
Running aws-sdk-php v3.275.6
PHP 7.4.33

```
Fatal error: Uncaught Error: Undefined class constant 'VERSION'in /html/app/library/composer/guzzlehttp/guzzle/src/functions.php: 136 

Stack trace: 
#0/html/app/library/composer/aws/aws-sdk-php/src/Handler/GuzzleV6/GuzzleHandler.php(40): GuzzleHttp\default_user_agent()
#1/html/app/library/composer/aws/aws-sdk-php/src/WrappedHttpHandler.php(88): Aws\Handler\GuzzleV6\GuzzleHandler->__invoke(Object(GuzzleHttp\Psr7\Request),
Array)
#2/html/app/library/composer/aws/aws-sdk-php/src/ClientSideMonitoring/AbstractMonitoringMiddleware.php(126): Aws\WrappedHttpHandler->__invoke(Object(Aws\Command),
Object(GuzzleHttp\Psr7\Request))
#3/html/app/library/composer/aws/aws-sdk-php/src/S3/PermanentRedirectMiddleware.php(43): Aws\ClientSideMonitoring\AbstractMonitoringMiddleware->__i in /html/app/library/composer/guzzlehttp/guzzle/src/functions.php on line 136
```

composer show guzzlehttp/guzzle:
```
name     : guzzlehttp/guzzle
descrip. : Guzzle is a PHP HTTP client library
keywords : client, curl, framework, http, http client, rest, web service
versions : * 6.5.8
type     : library
license  : MIT License (MIT) (OSI approved) https://spdx.org/licenses/MIT.html#licenseText
homepage : http://guzzlephp.org/
source   : [git] https://github.com/guzzle/guzzle.git a52f0440530b54fa079ce76e8c5d196a42cad981
dist     : [zip] https://api.github.com/repos/guzzle/guzzle/zipball/a52f0440530b54fa079ce76e8c5d196a42cad981 a52f0440530b54fa079ce76e8c5d196a42cad981
path     : /html/app/library/composer/guzzlehttp/guzzle
names    : guzzlehttp/guzzle

autoload
files
psr-4
GuzzleHttp\ => src/

requires
ext-json *
guzzlehttp/promises ^1.0
guzzlehttp/psr7 ^1.9
php >=5.5
symfony/polyfill-intl-idn ^1.17

requires (dev)
ext-curl *
phpunit/phpunit ^4.8.35 || ^5.7 || ^6.4 || ^7.0
psr/log ^1.1

suggests
psr/log Required for using the Log middleware
```

[a comment by an automated account is not shown]
